### DISCUSSION TITLE
Command line tool for composite colour

### DISCUSSION THREAD

**Discussion by matteo-tonini** [cssbattle/forum#151] [Show and tell]
Hi everyone :) ! 
I am fairly new at CSS battles, but I have become addicted to it quite fast.

In the last few days I have coded a [composite-golf](https://github.com/MTxx87/composite-golf) (https://github.com/MTxx87/composite-golf) command line tool.
I thought would be good to share it with the community.

It can be used to compute a set of possible overlay colours, given a desired colour and a background, like so:

`node . overlay 'rgba(69,60,250,1)' 'rgb(68,159,171)'  --alpha=100`

This would return a set of overlay colours that result in `rgb(68,159,171)` on background `rgba(69,60,250,1)`.

The math I used gives good results, but sometime the resulting colour is slightly off compared to what it should be.
If someone wants to check the code and improve the formula, please do so.
Also comments and suggestions are very welcome.

/Matteo

**Comment by chinchang** [comment]
woah!! This is crazy good tool Matteo! Thanks for sharing with the community 🙌🏽

**Comment by rafehimself** [comment]
Welcome, I guess we are from the same year 🤣

**Comment by matteo-tonini** [comment]
I pushed an update this morning.
Now the results gets highlighted if it is possible to use the shortened HEX code.

<img width="928" alt="Screen Shot 2021-10-31 at 13 32 36" src="https://user-images.githubusercontent.com/12138301/139583404-abe41359-6106-4864-a027-20eb1383a109.png">

**Reply by rafehimself** [reply]
and does it work for u ? I didn't try your tool, but so far I haven't success in shortening colors, machine doesn't accept it as 💯 
for example #53 
background:#4F77FF -> overlay: #9AD5FF
#9dfe, #adfd, #aefc, #bffa, #bffb all that shorten version doesn't give 💯 match

**Reply by matteo-tonini** [reply]
No, I have never had success in shortening the color codes. I think it is just good to know that it is mathematically impossible to shorten the color code, instead of guessing and try to get as close as possible by trial and error.

The formula for computing the colors often returns numbers with decimals that have to be rounded in order to be converted to HEX.
I have pushed an update that adds another column with how much error was introduced by the rounding.

if you're lucky enough to get a green coloured code with no error (empty column), 
that's your best chance that it will be accepted by the machine :) 

<img width="915" alt="Screen Shot 2021-11-01 at 21 28 04" src="https://user-images.githubusercontent.com/12138301/139737660-0ee6a259-3105-479d-a20d-4dd04e21eef6.png">
